src/components/include/cos_kernel_api.h

```diff
@@ -30,7 +30,7 @@ struct cos_meminfo {
 /* Component captbl/pgtbl allocation information */
 struct cos_compinfo {
 	/* capabilities to higher-order capability tables (or -1) */
-	capid_t pgtbl_cap, captbl_cap, comp_cap;
+	capid_t pgtbl_cap, captbl_cap, comp_cap, local_pgtbl_cap;
```

Comment:
Yes, I agree. If local memory is a common case, it is better so keep current local memory management code unchanged. As you said below, if we only have one new page table for global memory, we can avoid all changes I made for local memory.
But I do not think we want to merge the code for global memory into the mainstream, right?
I can modify the code in my own branch.